src/lib/utils/os_utils.cpp

```diff
@@ -206,32 +206,28 @@ uint64_t OS::get_cpu_cycle_counter()
 
 size_t OS::get_cpu_total()
    {
-   size_t tt = 1;
 #if defined(BOTAN_TARGET_OS_HAS_POSIX1) && defined(_SC_NPROCESSORS_CONF)
-   long res;
-   if ((res = ::sysconf(_SC_NPROCESSORS_CONF)) <= 0)
-     {
-          return 1;
-     }
-     tt = static_cast<size_t>(res);
-#elif defined(BOTAN_TARGET_OS_HAS_THREADS)
-     tt = static_cast<size_t>(std::thread::hardware_concurrency());
-#endif
-     return tt;
+   const long res = ::sysconf(_SC_NPROCESSORS_CONF);
```

Comment:
I see you rarely use `auto` (maybe for clarity's sake) so I just followed but nothing against.

Reply:
Yeah in general I only use `auto` when the type name is either very long (`std::map<...>::const_iterator`) or is actually untypable (lambda, based on template param, ...).